code/modules/projectiles/guncrafting/attachment.dm

```diff
@@ -0,0 +1,29 @@
+/obj/item/weapon/gun_attachment
+	name = "attach"
+	icon = 'icons/obj/guncrafting/main.dmi'
+	var/gun_type = BOTHTYPES
+	var/image/my_overlay
+	var/uses_overlay = TRUE
+	var/not_okay = /obj/item/weapon/gun_attachment
+
+/obj/item/weapon/gun_attachment/proc/can_attach(var/obj/item/weapon/gun/owning_gun) // Sorry but if I don't put a var in front of these it gives me duplicate definition errors pointing to here
```

Comment:
No, we wen't over this, ```obj/item/weapon/gun/owning_gun``` your comment is wrong because you left a ```/``` at the beginning.